snuba/clusters/cluster.py

```diff
@@ -137,37 +141,46 @@ def __init__(
         super().__init__(storage_sets)
         if not single_node:
             assert cluster_name
-        self.__host = host
-        self.__port = port
+
+        self.__query_node = ClickhouseNode(host, port)
         self.__http_port = http_port
```

Comment:
Why isn't the http port part of the node now? It seems to me it should not be treated differently than the TCP port,  or is there a specific reason ?

Reply:
The reason it's not part of the ClickhouseNode is because we only have the HTTP port of the query node and not any of the others. It's provided via configuration and we don't use it for any other nodes, and don't have a way to discover it via system.clusters. I can move it into the ClickhouseNode, but it will be None in all cases except the query node.

Reply:
ok, I guess it does not really make much difference as long as we only structurally allow one query node. Should that change we can move it.